pallets/aliases/src/lib.rs

```diff
@@ -0,0 +1,379 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+extern crate core;
+
+use codec::{Compact, Decode, Encode};
+use frame_support::traits::tokens::fungible;
+pub use pallet::*;
+use pallet_clusters::Clusters;
+use pallet_fragments::GetInstanceOwnerParams;
+use pallet_protos::{Proto, ProtoOwner, Protos};
+use sp_clamor::{Hash128, Hash256, InstanceUnit};
+use sp_std::vec::Vec;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(test)]
+mod dummy_data;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+/// Enum that indicates the types of target assets linked to an alias
+#[derive(Encode, Decode, Clone, scale_info::TypeInfo, Debug, PartialEq)]
+pub enum LinkTarget<TAccountId> {
+	Proto(Hash256),
+	Fragment { definition_hash: Vec<u8>, edition_id: InstanceUnit, copy_id: InstanceUnit }, // struct variant allows a nicer UX on polkadotJS
+	Account(TAccountId),
+	Cluster(Hash128),
+}
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::{
+		pallet_prelude::*,
+		sp_runtime::{traits::Zero, SaturatedConversion},
+	};
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	/// Configure the pallet by specifying the parameters and types on which it depends.
+	#[pallet::config]
+	pub trait Config:
+		frame_system::Config
+		+ pallet_balances::Config
+		+ pallet_protos::Config
+		+ pallet_fragments::Config
+		+ pallet_clusters::Config
+	{
+		/// Because this pallet emits events, it depends on the runtime's definition of an event.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// The price (in native NOVA) for a Namespace
+		#[pallet::constant]
+		type NamespacePrice: Get<u128>;
+
+		/// The max size of namespace string
```

Comment:
This should be namespace and name , both elements need a limit

Reply:
Updated the comment as this is used for both.